static/src/javascripts/projects/common/modules/identity/email-preferences.js

```diff
@@ -0,0 +1,119 @@
+// This is a workaround for the email preferences page https://profile.thegulocal.com/email-prefs
+// We want to submit subscribe/unsubscribe requests without a full page refresh
+// Hopefully this will be short-lived; if it is still alive in 2017, git blame and cry
+
+define([
+    'bean',
+    'reqwest',
+    'fastdom',
+    'common/utils/$'
+], function (
+    bean,
+    reqwest,
+    fastdom,
+    $
+) {
+    function reqwestEmailSubscriptionUpdate(buttonEl) {
+        bean.on(buttonEl, 'click', function () {
+            buttonEl.disabled = true;
+            buttonEl.innerHTML = 'Loading...';
+            var formQueryString = generateFormQueryString(buttonEl);
+            reqwest({
+                url: '/email-prefs',
+                method: 'POST',
+                data: formQueryString,
+                error: function () {
+                    renderErrorMessage(buttonEl);
+                },
+                success: function (response) {
+                    var isSubscribed = false;
+                    if (response && response.subscriptions && response.subscriptions.length) {
+                        isSubscribed = true;
+                    }
+                    updateButton(buttonEl, isSubscribed);
+                }
+            });
+        });
+    }
+
+    function enhanceEmailPreferences() {
+        $.forEachElement('.email-subscription__button', reqwestEmailSubscriptionUpdate);
+        $.forEachElement('.save__button', reqwestEmailSubscriptionUpdate);
+    }
+
+    function encodeFormData(csrfToken, buttonVal, htmlPreference) {
+        return 'csrfToken=' + encodeURIComponent(csrfToken) + '&' +
+        'addEmailSubscription=' + encodeURIComponent(buttonVal) + '&' +
+        'htmlPreference=' + encodeURIComponent(htmlPreference);
+    }
+
+    function renderErrorMessage(buttonEl) {
+        return fastdom.write(function () {
+            clearErrorMessages();
+            var errorMessage = $.create(
+                '<div class="form__error">' +
+                    'Sorry, an error has occurred, please refresh the page and try again' +
+                '</div>'
+            );
+            $(errorMessage).insertAfter(buttonEl.parentNode);
+        });
+    }
+
+    function clearErrorMessages() {
+        if ($('.form__error')) {
+            $.forEachElement('.form__error', function (errorEl) {
+                errorEl.parentNode.removeChild(errorEl);
+            });
+        }
+    }
+
+    function updateSubscriptionButton(buttonEl, isSubscribed) {
+        var buttonVal = buttonEl.value;
+        if (isSubscribed) {
+            fastdom.write(function () {
+                buttonEl.value = 'unsubscribe-' + buttonVal;
+                buttonEl.innerHTML = 'Unsubscribe';
+                $($.ancestor(buttonEl, 'email-subscription')).addClass('email-subscription--subscribed');
+                buttonEl.disabled = false;
+            });
+        } else if (!isSubscribed) {
```

Comment:
this is actually redundant - `!isSubscribed` will always have the opposite value to `isSubscribed` so it's the same as `else`. I would ditch the `renderErrorMessage` bit and make this an `else`